DemoApp/middlewares/upload.js

```diff
@@ -0,0 +1,47 @@
+const util = require("util");
+const multer = require("multer");
+const maxSize = 2 * 1024 * 1024;
+const path = require('path');
+const uuid = require('uuid').v4;
+const Report=require('../models/Report');
+let storage = multer.diskStorage({
+  destination: (req, file, cb) => {
+    cb(null,"upload");
+    
+
+  },
+  filename: (req, file, cb) => {
+    console.log("dddddd"+file.originalname)
+    const ext=path.extname(file.originalname)
+    const id=uuid();
+    const filepath=`documents/${id}${ext}`; 
+    console.log(filepath)
+    Report.create({filepath})
```

Comment:
Again, then chaining, I strongly recommend to switch to async/await and try,catch syntax.